drivers/led/led_dac.c

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2025 Prevas A/S
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT dac_leds
+
+#include <errno.h>
+#include <stdint.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/dac.h>
+#include <zephyr/drivers/led.h>
+#include <zephyr/sys/math_extras.h>
+
+struct led_dac_leds {
+	const struct device *dac;
+	struct dac_channel_cfg chan_cfg;
+	uint32_t dac_max;
+	uint32_t dac_min;
```

Comment:
I'd suggest the following to avoid confusion with the actual dac max output value (e.g. 3.3V).

```suggestion
	uint32_t dac_max_brightness;
	uint32_t dac_min_brightness;
```